tests/tests.py

```diff
@@ -0,0 +1,58 @@
+import unittest
+from sudo_root.stegano import LSBExtractor
+from sudo_root.misc import zxing
+from sudo_root.crypto import Lcg
+
+
+class TestLSBExtractor(unittest.TestCase):
+
+    def test_all_lsb(self):
+        img_path = "./test_assets/stegano/lamiri_sba.png"
+
+        lsb = LSBExtractor(img_path)
+        out = lsb.all_lsb(40)
+
+        msg = ""
+        for i in range(0, len(out), 8):
+            msg += chr(int(out[i:i+8], 2))
+
+        self.assertEqual(msg, "Hello World !!!")
+
+    def test_all_cycle(self):
+        img_path = "./test_assets/stegano/esi_sba_lsb_cycle.png"
+
+        lsb = LSBExtractor(img_path)
+        out = lsb.cycle_lsb(192)
+
+        msg = ""
+        for i in range(0, len(out), 8):
+            msg += chr(int(out[i:i+8], 2))
+
+        self.assertEqual(msg, "LSB cycle hidden message")
+
+
+class TestZxing(unittest.TestCase):
```

Comment:
does the tests here works for you?

Reply:
the test should work on both python versions 2 and 3, but the "Zxing" module use "HTMLParser" module which is available on python2 only.
we should re-implement the "Zxing" module and use "lxml" module to parse the html.

Reply:
on it